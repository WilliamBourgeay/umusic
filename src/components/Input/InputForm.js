// @flow

import React from 'react';
import TextField from '@material-ui/core/TextField';

import './InputForm.css';

type Props = {
  id: string,
  name?: string,
  type?: string,
  onEnterKey?: Function,
  className?: string,
  placeholder?: string,
};

type State = {
  value: string,
};

class InputForm extends React.Component<Props, State> {
  static defaultProps = {
    name: '',
    type: 'text',
    onEnterKey: undefined,
    className: '',
    placeholder: '',
  };

  state = {
    value: '',
  };

  handleKeyPress = (event: any) => {
    const { onEnterKey } = this.props;
    const { value } = this.state;

    if (event.key !== 'Enter') return;

    event.preventDefault();

    if (onEnterKey) {
      onEnterKey(value);
      this.setState({ value: '' });
    }
  };

  handleChange = (event: any) => {
    this.setState({ value: event.target.value });
  };

  render() {
    const { id, name, type, className, placeholder } = this.props;
    const { value } = this.state;

    return (
      <TextField
        id={id}
        label={name}
        type={type}
        margin="normal"
        onKeyPress={this.handleKeyPress}
        onChange={this.handleChange}
        className={className}
        placeholder={placeholder}
        value={value}
      />
    );
  }
}

export default InputForm;
