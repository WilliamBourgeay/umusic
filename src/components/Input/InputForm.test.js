import * as React from 'react';
import renderer from 'react-test-renderer';

import InputForm from './InputForm';

describe('InputForm', () => {
  it('should render an Input Form', () => {
    const tree = renderer.create(<InputForm id="test" />);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should render an Input Form with name', () => {
    const tree = renderer.create(<InputForm id="test" name="name" />);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should render an Input Form with type', () => {
    const tree = renderer.create(<InputForm id="test" type="email" />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
