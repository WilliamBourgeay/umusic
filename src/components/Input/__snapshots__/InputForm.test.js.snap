// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputForm should render an Input Form 1`] = `
<div
  aria-describedby={undefined}
  className="MuiFormControl-root-1 MuiFormControl-marginNormal-2"
  minlength={undefined}
  onKeyPress={[Function]}
>
  
  <div
    className="MuiInputBase-root-18 MuiInput-root-5 MuiInput-underline-9 MuiInputBase-formControl-19 MuiInput-formControl-6"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoComplete={undefined}
      autoFocus={undefined}
      className="MuiInputBase-input-28 MuiInput-input-13"
      defaultValue={undefined}
      disabled={false}
      id="test"
      name={undefined}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={undefined}
      onKeyUp={undefined}
      placeholder=""
      readOnly={false}
      required={false}
      rows={undefined}
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`InputForm should render an Input Form with name 1`] = `
<div
  aria-describedby={undefined}
  className="MuiFormControl-root-1 MuiFormControl-marginNormal-2"
  minlength={undefined}
  onKeyPress={[Function]}
>
  <label
    className="MuiFormLabel-root-42 MuiInputLabel-root-35 MuiInputLabel-formControl-36 MuiInputLabel-animated-39"
    data-shrink={false}
    htmlFor="test"
  >
    name
  </label>
  <div
    className="MuiInputBase-root-18 MuiInput-root-5 MuiInput-underline-9 MuiInputBase-formControl-19 MuiInput-formControl-6"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoComplete={undefined}
      autoFocus={undefined}
      className="MuiInputBase-input-28 MuiInput-input-13"
      defaultValue={undefined}
      disabled={false}
      id="test"
      name={undefined}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={undefined}
      onKeyUp={undefined}
      placeholder=""
      readOnly={false}
      required={false}
      rows={undefined}
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`InputForm should render an Input Form with type 1`] = `
<div
  aria-describedby={undefined}
  className="MuiFormControl-root-1 MuiFormControl-marginNormal-2"
  minlength={undefined}
  onKeyPress={[Function]}
>
  
  <div
    className="MuiInputBase-root-18 MuiInput-root-5 MuiInput-underline-9 MuiInputBase-formControl-19 MuiInput-formControl-6"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoComplete={undefined}
      autoFocus={undefined}
      className="MuiInputBase-input-28 MuiInput-input-13 MuiInputBase-inputType-31 MuiInput-inputType-16"
      defaultValue={undefined}
      disabled={false}
      id="test"
      name={undefined}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={undefined}
      onKeyUp={undefined}
      placeholder=""
      readOnly={false}
      required={false}
      rows={undefined}
      type="email"
      value=""
    />
  </div>
</div>
`;
